sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/Monitor.java

```diff
@@ -39,20 +39,15 @@
   private class MonitorDoFn extends DoFn<T, T> {
     final Counter elementCounter = Metrics.counter(name, prefix + ".elements");
     final Counter bytesCounter = Metrics.counter(name, prefix + ".bytes");
-    final Distribution startTime = Metrics.distribution(name, prefix + ".startTime");
-    final Distribution endTime = Metrics.distribution(name, prefix + ".endTime");
-    final Distribution startTimestamp = Metrics.distribution(name, prefix + ".startTimestamp");
-    final Distribution endTimestamp = Metrics.distribution(name, prefix + ".endTimestamp");
+    final Distribution systemTimestamp = Metrics.distribution(name, prefix + ".systemTimestamp");
+    final Distribution eventTimestamp = Metrics.distribution(name, prefix + ".eventTimestamp");
 
     @ProcessElement
     public void processElement(ProcessContext c) {
       elementCounter.inc();
       bytesCounter.inc(c.element().sizeInBytes());
-      long now = System.currentTimeMillis();
-      startTime.update(now);
-      endTime.update(now);
-      startTimestamp.update(c.timestamp().getMillis());
-      endTimestamp.update(c.timestamp().getMillis());
+      systemTimestamp.update(System.currentTimeMillis());
```

Comment:
agree with your simplification for these reasons:
- for each element processed, the distribution metrics start and end will be updated with the same value and, as the distribution reports min, max and average, and, as in _MetricsReader_ we are requesting the minimum and the maximum, there is no impact of having a single distribution
- the methods in _MetricsReader_ use a SplitIterator on the distributions that is not immutable for the underlying source (which could explain that there was 2 distributions created) but it does only reading (by calling _getMin()_ and _getMax()_) so there will be no side effect on the underlying distribution.